pypowerwall/tedapi/vitals_dictionary.py

```diff
@@ -0,0 +1,517 @@
+
+    def get_vitals(self):
+        return {
+            **self.get_header(),
+            **self.get_neurio(),
+            **self.get_pvac(),
+            **self.get_pvs(),
+            **self.get_ststsm(),
+            **self.get_tepinv(),
+            **self.get_tepod(),
+            **self.get_tesla(),
+            **self.get_tesync(),
+            **self.get_tethc(),
+        }
+
+    def get_header(self):
+        header = {}
+        header["VITALS"] = {
+            "text": "Device vitals generated from Tesla Powerwall Gateway TEDAPI",
+            "timestamp": time.time(),
+            "gateway": self.gw_ip,
+            "pyPowerwall": __version__,
+        }
+        return header
+
+    def get_meters(self):
+        meter_config = {}
+        if "meters" in self.config:
+            # Loop through each meter and use device_serial as the key
+            for meter in self.config['meters']:
+                if meter.get('type') == "neurio_w2_tcp":
+                    device_serial = lookup(meter, ['connection', 'device_serial'])
+                    if device_serial:
+                        # Check to see if we already have this meter in meter_config
+                        if device_serial in meter_config:
+                            cts = meter.get('cts', [False] * 4)
+                            if not isinstance(cts, list):
+                                cts = [False] * 4
+                            for i, ct in enumerate(cts):
+                                if ct:
+                                    meter_config[device_serial]['cts'][i] = True
+                                    meter_config[device_serial]['location'][i] = meter.get('location', "")
+                        else:
+                            # New meter, add to meter_config
+                            cts = meter.get('cts', [False] * 4)
+                            if not isinstance(cts, list):
+                                cts = [False] * 4
+                            location = meter.get('location', "")
+                            meter_config[device_serial] = {
+                                "type": meter.get('type'),
+                                "location": [location] * 4,
+                                "cts": cts,
+                                "inverted": meter.get('inverted'),
+                                "connection": meter.get('connection'),
+                                "real_power_scale_factor": meter.get('real_power_scale_factor', 1)
+                            }
+        return meter_config
+
+    def get_neurio(self):
+        # Build meter Lookup if available
+        meter_config = self.get_meters()
+        neurio = {}
+        c = 1000
+        # Loop through each Neurio device serial number
+        for n in lookup(self.status, ['neurio', 'readings']) or {}:
+            # Loop through each CT on the Neurio device
+            sn = n.get('serial', str(c))
+            cts = {}
+            c = c + 1
```

Comment:
`c` can be computed using enumerate as well in the bigger, outer loop.

Reply:
Ditto this one, without the hardware to test this I'm inclined to leave it as it was in the original file